src/main/java/com/iexec/core/sms/SmsService.java

```diff
@@ -55,24 +54,31 @@ public SmsService(SmsClientProvider smsClientProvider, IexecHubAbstractService i
      * <p>
      * If any of these conditions is wrong, then the {@link SmsClient} is considered to be not-ready.
      *
-     * @param chainDealId ID of the on-chain deal related to the task to execute.
      * @param chainTaskId ID of the on-chain task.
+     * @param tag Tag of the deal
      * @return {@literal true} if previous conditions are met, {@literal false} otherwise.
      */
-    public boolean isSmsClientReady(String chainDealId, String chainTaskId) {
-        try {
-            final Optional<ChainDeal> chainDeal = iexecHubService.getChainDeal(chainDealId);
-            if (chainDeal.isEmpty()) {
-                log.error("No chain deal for given ID [chainDealId: {}]", chainDealId);
-                return false;
-            }
-            final SmsClient smsClient = smsClientProvider.getOrCreateSmsClientForUninitializedTask(chainDeal.get(), chainTaskId);
-            final TeeEnclaveProvider teeEnclaveProviderForDeal = TeeUtils.getTeeEnclaveProvider(chainDeal.get().getTag());
-            return checkSmsTeeEnclaveProvider(smsClient, teeEnclaveProviderForDeal, chainTaskId);
-        } catch (SmsClientCreationException e) {
-            log.error("SmsClient is not ready [chainTaskId: {}]", chainTaskId, e);
-            return false;
+    public String getVerifiedSmsUrl(String chainTaskId, String tag) {
+        final TeeEnclaveProvider teeEnclaveProviderForDeal = TeeUtils.getTeeEnclaveProvider(tag);
+        if(teeEnclaveProviderForDeal == null){
+            return "";
         }
+        String smsUrl = retrieveSmsUrl(teeEnclaveProviderForDeal);
+        final SmsClient smsClient = smsClientProvider.getSmsClient(smsUrl);
+        if(!checkSmsTeeEnclaveProvider(smsClient, teeEnclaveProviderForDeal, chainTaskId)){
+            return "";
+        }
+        return smsUrl;
+    }
+
+    private String retrieveSmsUrl(TeeEnclaveProvider teeEnclaveProvider) {
+        String smsUrl = "";
+        if(TeeEnclaveProvider.SCONE.equals(teeEnclaveProvider)){
+            smsUrl = registryConfiguration.getSconeSms();
+        } else if(TeeEnclaveProvider.GRAMINE.equals(teeEnclaveProvider)){
+            smsUrl = registryConfiguration.getGramineSms();
+        }
+        return smsUrl;
```

Comment:
Why not throwing an exception or returning an Optional instead of an empty String in case no matching `TeeEnclaveProvider`? We should also log this error.

Reply:
See https://github.com/iExecBlockchainComputing/iexec-core/pull/523#discussion_r977741179 :)